.github/workflows/ci.yml

```diff
@@ -0,0 +1,57 @@
+name: Unit Tests
+
+on: [push]
+
+jobs:
+
+  pytest:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:13
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres  # pragma: allowlist secret
+          POSTGRES_DB: indexd
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          # Maps tcp port 5432 on service container to the host
+          - 5432:5432
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.10.0
+    - uses: actions/cache@preview
```

Comment:
Not entirely sure how/where this cache action is used or if it is needed

Reply:
this is caching the virtual env so if the same files come in again without any dependency changes, it doesn't have to reinstall everything - so we should keep it